Source/WebCore/rendering/svg/RenderSVGResourceClipper.cpp

```diff
@@ -160,14 +151,28 @@ bool RenderSVGResourceClipper::applyClippingToContext(GraphicsContext& context,
     // Determine scale factor for the clipper. The size of intermediate ImageBuffers shouldn't be bigger than kMaxFilterSize.
     ImageBuffer::sizeNeedsClamping(objectBoundingBox.size(), scale);
 
-    if (!clipperData.isValidForGeometry(objectBoundingBox, clippedContentBounds, absoluteTransform)) {
+    return { objectBoundingBox, clippedContentBounds, scale, effectiveZoom };
+}
+
+bool RenderSVGResourceClipper::applyClippingToContext(GraphicsContext& context, RenderElement& renderer, const FloatRect& objectBoundingBox, const FloatRect& clippedContentBounds, float effectiveZoom)
+{
+    auto& clipperData = *m_clipperMap.ensure(&renderer, [&]() {
+        return makeUnique<ClipperData>();
+    }).iterator->value;
+
+    LOG_WITH_STREAM(SVG, stream << "RenderSVGResourceClipper " << this << " applyClippingToContext: renderer " << &renderer << " objectBoundingBox " << objectBoundingBox << " clippedContentBounds " << clippedContentBounds << " (existing image buffer " << clipperData.imageBuffer.get() << ")");
+
+    AffineTransform animatedLocalTransform = clipPathElement().animatedLocalTransform();
+
+    if (!clipperData.imageBuffer && pathOnlyClipping(context, animatedLocalTransform, objectBoundingBox, effectiveZoom))
+        return true;
+
+    if (clipperData.invalidate(computeInputs(renderer, objectBoundingBox, clippedContentBounds, effectiveZoom))) {
```

Comment:
I am confused here. We used to regenerate the maskImage if absoluteTransform changes. Now we look at the changes in effectiveZoom. effectiveZoom does not affect the size of the maskImage but absoluteTransform can affect the size of the maskImage since we get the scale out of it

```
    // Ignore 2D rotation, as it doesn't affect the size of the mask.
    FloatSize scale(absoluteTransform.xScale(), absoluteTransform.yScale());
```

Reply:
We store both the `effectiveZoom` and the scale extracted from the `absoluteTransform` in the `Inputs`, so we invalidate the mask image if either of those change.